experimental/displaynames/src/displaynames.rs

```diff
@@ -155,17 +146,27 @@ impl LanguageDisplayNames {
         ]
     );
 
-    /// Returns the locale specific display name of a language for a given string.
-    /// This function is locale-sensitive.
-    pub fn of(&self, code: &str) -> Option<&str> {
-        let key = UnvalidatedStr::from_str(code);
+    /// Returns the display name of a locale.
+    pub fn of(&self, langid: &Locale) -> Option<&str> {
```

Comment:
I think we should take LanguageIdentifier here

Reply:
UTS35 defines the algorithm for Unicode extensions: https://www.unicode.org/reports/tr35/tr35-general.html#Display_Name_Elements:~:text=en%2Du%2Dnu%2Dthai%2Dca%2Dislamic%2Dcivil

Do we want to implement that?

Reply:
How about we use `Language` here to just do a tag lookup like with `Region`, and then have a more powerful method that works on `Locale`? This would check for special cases first (stored in another key), and if it doesn't get one it would assemble the display string according to the algorithm.

Reply:
Yes, we should implement the full UTS 35, and the implementation of it should be called LocaleDisplayNames.

What we currently have here only supports the data in `languages.json` (LanguageDisplayNames) and `regions.json` (RegionDisplayNames). LocaleDisplayNames will need to glue those together and add in the stuff about variants and extension keywords.

Actually I think LanguageDisplayNames is a fairly low-level utility, because callers need to implement their own fallbacking. For example, `zh-Hant-TW` does not have its own entry, but `zh-Hant` does. However, `pt-BR` has its own entry. We may want to add a configuration (perhaps by default) to LanguageDisplayNames to perform that type of fallbacking on its own.